src/main/java/com/gotocompany/depot/maxcompute/util/TypeInfoUtils.java

```diff
@@ -0,0 +1,24 @@
+package com.gotocompany.depot.maxcompute.util;
+
+import com.aliyun.odps.type.ArrayTypeInfo;
+import com.aliyun.odps.type.PrimitiveTypeInfo;
+import com.aliyun.odps.type.StructTypeInfo;
+import com.aliyun.odps.type.TypeInfo;
+
+public class TypeInfoUtils {
+    public static boolean isPrimitiveType(TypeInfo typeInfo) {
+        return typeInfo instanceof PrimitiveTypeInfo;
+    }
+
+    public static boolean isStructArrayType(TypeInfo typeInfo) {
+        return typeInfo instanceof ArrayTypeInfo && ((ArrayTypeInfo) typeInfo).getElementTypeInfo() instanceof StructTypeInfo;
+    }
+
+    public static boolean isPrimitiveArrayType(TypeInfo typeInfo) {
+        return typeInfo instanceof ArrayTypeInfo && ((ArrayTypeInfo) typeInfo).getElementTypeInfo() instanceof PrimitiveTypeInfo;
```

Comment:
These utility methods perform type checks repeatedly without caching. For frequently accessed types, this could impact performance. Consider implementing a caching mechanism.